ex02.exs

```diff
@@ -1,6 +1,24 @@
 
 defmodule Ex02 do
 
+  @global_state __MODULE__
+
+  def new_counter( init_value \\ 0 ) do
+    Agent.start(fn -> init_value end)
+  end
+
+  def next_value( { _, counter } ) do    
```

Comment:
## -3 style

It took me a little while to work out why you were pattern matching here, then I worked out you simply took the return value from Agent.start and passed it back into next_value().

It would be better encapsulation to do the pattern match when you create the agent in `new_counter` and then simply pass out the agent's pid.